sympy/core/expr.py

```diff
@@ -3668,9 +3659,7 @@ def round(self, n=None):
                     'Expected a number but got %s:' % func_name(x))
         elif x in (S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity):
             return x
-        if not x.is_extended_real:
```

Comment:
This is not Python-2 related and should be retained.

Reply:
Sorry, I had some confusion about how to run the tests and thought that `sympy.test("sympy/core/expr.py")` passing meant the code was working when I should have been testing `sympy.test("sympy/core/tests/test_expr.py")`